### ISSUE TITLE
Future Option Parameter Flags and Behaviours

### ISSUE THREAD

**Issue by Ink230** [Ink230/DirectoryPredictor#2]
## Description
There are a number of behaviours the DirectoryPredictor should be configurable with. Common and fringe use cases should be accommodated for accessibility and a mature module. More option parameters than listed are wanted.

## Potential Option Parameters
* ~~-ResultsLimit [integer]~~
  * ~~Specifies how many results to return~~
  * ~~Is limited up to 10 by PSReadLine customizability~~
* -CaseBehaviour [Ignore] | [Respect] | [???]
  * Ignore: Current behaviour lowercases everything and ignores cases
  * Respect: Displays any combination of case usage and respects it in the suggestion display
    * A lot of minor issues with files like hEllOWorlD where the user's search is HELLO or HEL or hEl
  * ???: Need input on what other behaviours may be appropriate
* ~~-IgnoreCommands [string]~~
  * ~~A list of commands that will not trigger the DirectoryPredictor~~
    * ~~Ex: -IgnoreCommands "ls,cd,clear"~~
  * ~~Comma separated string, unless a better way to pass multiple cmds as strings exists~~
 * -DirectoryDepth [integer]
   * How many subdirectories to parse through
     * Depth 1 is the current directory
     * Depth 2 is all files in the folders of the current directory
   * Potential performance considerations
 * -Folders [Include] | [Exclude]
   * Default behaviour will be to exclude folders
   * Open questions
     * Should folders be mixed in with files or separated at the top or bottom of the list
     * Should files or folders take priority
     * Should folders be marked as a folder somehow
     * These are all probably additional parameters and some may not be possible with PSReadLine
   * Potentially not necessary
   
## Blocking Issues
~~#1 - Multiple Options Parameters Cause All Previous to be Ignored~~

**Comment by Ink230** [comment]
https://github.com/Ink230/DirectoryPredictor/issues/1 is resolved

**Comment by Ink230** [comment]
ResultsLimit and -IgnoreCommands functionality added with commits https://github.com/Ink230/DirectoryPredictor/commit/06f4dcf6457ad9ae6d3473e670bacc545d5bc0eb and https://github.com/Ink230/DirectoryPredictor/commit/f8d50580b620a7ea185c56c24e9fb3a5e6951309 https://github.com/Ink230/DirectoryPredictor/commit/db08456f5e8155acfa33b72f7529123eb15159e6, respectively.

**Comment by Ink230** [comment]
The originally listed flags seem too niche with many complicated fringe cases to account for. Only upon popular demand will they most likely be added.

Other option cmdlet flags are always open for suggestion in a new Issue thread.
